crates/astria-composer/src/searcher/mod.rs

```diff
@@ -148,34 +160,34 @@ impl Searcher {
     }
 
     /// Serializes and signs a sequencer tx from a rollup tx.
-    fn bundle_pending_tx(&mut self, tx: collector::Transaction) {
-        let collector::Transaction {
+    fn bundle_pending_tx(&mut self, tx: collector::RollupTransaction) {
+        let collector::RollupTransaction {
             rollup_id,
             inner: rollup_tx,
         } = tx;
+        let max_bundle_sz = self.max_bundle_sz;
 
         // rollup transaction data serialization is a heavy compute task, so it is spawned
         // on tokio's blocking threadpool
         self.conversion_tasks.spawn_blocking(move || {
             let data = rollup_tx.rlp().to_vec();
-            let seq_action = Action::Sequence(SequenceAction {
+            let seq_action = SequenceAction {
                 rollup_id,
                 data,
-            });
+            };
 
-            vec![seq_action]
-        });
-    }
+            let seq_action_sz = seq_action.data.len() + ROLLUP_ID_LEN;
 
-    async fn handle_bundle_execution(&self, bundle: Vec<Action>) {
-        // send bundle to executor
-        if let Err(e) = self.bundle_tx.send(bundle).await {
-            error!(
-                error.message = %e,
-                error.cause_chain = ?e,
-                "failed to send bundle to executor",
-            );
-        }
+            if seq_action_sz > max_bundle_sz {
```

Comment:
This logic is now effectively duplicated here and in your bundle factory. Better have the bundle factory reject the bundle when you try to push it in